server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -178,40 +178,30 @@ private void handleCreate(CreationMessageEntry create, Builder responseBuilder,
     }
     
     private void performCreate(CreationMessageEntry entry, Builder responseBuilder, MailboxSession session) throws MailboxException, InvalidMailboxForCreationException, MessagingException, AttachmentsNotFoundException {
-        if (isAppendToMailboxWithRole(Role.DRAFTS, entry.getValue(), session)) {
+        if (isDraft(entry.getValue())) {
             saveDraft(entry, responseBuilder, session);
         } else if (isAppendToMailboxWithRole(Role.OUTBOX, entry.getValue(), session)) {
             sendMailViaOutbox(entry, responseBuilder, session);
         } else {
+            if (isAppendToMailboxWithRole(Role.DRAFTS, entry.getValue(), session)) {
```

Comment:
do we need to keep this?

Reply:
Yes we need to keep this: https://james.open-paas.org/job/workflow/274/console

The idea is to provide an accurate error message when the intent of the user is to save a draft.

Scenario : Saving a message in Draft mailbox without $Draft flag
Expected error: keywords: "Expecting draft messages to have $Draft keyword"
Without that code: mailboxIds invalid: "The only implemented feature is sending via outbox and draft saving"

I will re-add it.